cogni-libs/src/main/java/zone/cogni/libs/sparqlservice/impl/StardogSparqlService.java

```diff
@@ -0,0 +1,158 @@
+              .body(new UrlEncodedFormEntity(Collections.singletonList(new BasicNameValuePair("update", updateQuery)), StandardCharsets.UTF_8))
+              .execute();
+      checkAndDiscardResponse(response);
+    }
+    catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
+  }
+
+  @Override
+  public boolean executeAskQuery(String askQuery) {
+    //pass default graph, method without default graph doesn't use the HttpContext !!!
+    try (QueryExecution queryExecution = QueryExecutionFactory.sparqlService(endpointUrl + "/query", askQuery, null, null, authenticationHttpContext)) {
+      //jena adds empty defaultGraph param to URL because defaultGraph is null but is a "value", stardog doesn't like that
+      ((QueryEngineHTTP) queryExecution).setDefaultGraphURIs(Collections.emptyList());
+      return queryExecution.execAsk();
+    }
+
+  }
+
+  @Override
+  public void upload(Model model, String graphUri) {
+    updateGraph(graphUri, model);
+  }
+
+  @Override
+  public void updateGraph(String graphUri, Model model) {
+    upload(model, graphUri, false);
+  }
+
+  @Override
+  public void replaceGraph(String graphUri, Model model) {
+    upload(model, graphUri, true);
+  }
+
+  private void upload(Model model, String graphUri, boolean replace) {
+    String graphStoreUrl = endpointUrl + "?graph=" + graphUri;
+    StringWriter writer = new StringWriter();
+    try {
+      model.write(writer, "ttl");
+      Response response = (replace ? Request.Put(graphStoreUrl) : Request.Post(graphStoreUrl))  //Put replaces the graph, Post adds data
+              .setHeader("Content-Type", "text/turtle;charset=utf-8")
+              .setHeader(authHeader)
+              .bodyByteArray(writer.toString().getBytes(), ContentType.create("text/turtle", StandardCharsets.UTF_8))
+              .execute();
+      checkAndDiscardResponse(response);
+    }
+    catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+
+  @Override
+  public <R> R executeSelectQuery(String query, Function<ResultSet, R> resultHandler) {
+    //pass default graph, method without default graph doesn't use the HttpContext !!!
+    try (QueryExecution queryExecution = QueryExecutionFactory.sparqlService(endpointUrl + "/query", query, null, null, authenticationHttpContext)) {
+      //jena adds empty defaultGraph param to URL because defaultGraph is null but is a "value", stardog doesn't like that
+      ((QueryEngineHTTP) queryExecution).setDefaultGraphURIs(Collections.emptyList());
+      return resultHandler.apply(queryExecution.execSelect());
+    }
+  }
+
+  @Override
+  public void dropGraph(String graphUri) {
+    String graphStoreUrl = endpointUrl + "?graph=" + graphUri;
+    try {
+      Response response = Request.Delete(graphStoreUrl)
+              .setHeader(authHeader)
+              .execute();
+      checkAndDiscardResponse(response);
+    }
+    catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
```

Comment:
a code block thingy that does something like this maybe? not sure how it works with checked exception though

```
try {
  Response response = supplier.get();
  checkAndDiscardResponse(response);
}
catch (IOException e) {
  throw new RuntimeException(e);
}  
```